src/layman/authz/__init__.py

```diff
@@ -119,6 +119,28 @@ def authorize_after_request_tmp(response):
     return decorated_function
 
 
+def check_and_authorize_uuid_publication_decorator(expected_publication_type=None):
```

Comment:
I prefer to call this method just `authorize_uuid_publication_decorator` as authorization is the main goal and it's not clear what `check` means in the name.

I think we will call this method with `expected_publication_type` argument set in the near future, so I prefer this argument to be mandatory. And possibly to be keyword argument, not positional. E.g.

```suggestion
def authorize_uuid_publication_decorator(*, expected_publication_type):
```